server/schedule/checker/rule_checker.go

```diff
@@ -266,6 +279,24 @@ func (c *RuleChecker) fixLooseMatchPeer(region *core.RegionInfo, fit *placement.
 		checkerCounter.WithLabelValues("rule_checker", "demote-voter-role").Inc()
 		return operator.CreateDemoteVoterOperator("fix-demote-voter", c.cluster, region, peer)
```

Comment:
Is it possible that we can complete the scheduling of adding a witness learner at the same time? The current implementation seems to demote a voter first and then set it as a witness then.

Reply:
Can be at the same time, the current implementation is separated for logic simplicity.